cups/file.c

```diff
@@ -769,6 +769,7 @@ cupsFileGetConf(cups_file_t *fp,	/* I  - CUPS file */
 	  ptr --;
 	}
 
+    return (buf);
 	*ptr = '\0';
       }
     }
```

Comment:
Additionally you should review every cupsFilePut() below in ```cupsAdminSetServerSettings()``` which add a comment - once we start supporting comments, we can't put new comments in.

Rephrase the comments mentioned in ```cupsFilePut()``` and put it to the proper place at ```conf/cupsd.conf.in```, f.e. when line contains "</Location>" and we are in /admin location:

The code:
```
  if (remote_admin)
          cupsFilePuts(temp, "  # Allow remote administration...\n");
  else
          cupsFilePuts(temp, "  # Restrict access to the admin pages...\n");
```

The default cupsd.conf.in:

```
# Restrict access to the admin pages...
<Location /admin>
  AuthType Default
  Require user @SYSTEM
  Order allow,deny
</Location>
```

so once you remove the cupsFilePut(), update the cupsd.conf.in to look like this f.e.:

```
# Access to the admin pages:
# - default action defined by Order', see 'man cupsd.conf'
# - use 'Allow'/'Deny' for configuring access
<Location /admin>
  AuthType Default
  Require user @SYSTEM
  Order allow,deny
</Location>
```

Reply:
`     else if (in_root_location &&
               (remote_admin >= 0 || remote_any >= 0 || share_printers >= 0))
      {
	wrote_root_location = 1;

	if (remote_admin > 0 && share_printers > 0)
          cupsFilePuts(temp, "  # Allow shared printing and remote "
	                     "administration...\n");
	else if (remote_admin > 0)
          cupsFilePuts(temp, "  # Allow remote administration...\n");
	else if (share_printers > 0)
          cupsFilePuts(temp, "  # Allow shared printing...\n");
	else if (remote_any > 0)
          cupsFilePuts(temp, "  # Allow remote access...\n");
	else
          cupsFilePuts(temp, "  # Restrict access to the server...\n");

        cupsFilePuts(temp, "  Order allow,deny\n");

	if (remote_admin > 0 || remote_any > 0 || share_printers > 0)
	{
	  if (remote_any >= 0)
	    cupsFilePrintf(temp, "  Allow %s\n", remote_any > 0 ? "all" : "@LOCAL");
	  else
	    cupsFilePrintf(temp, "  Allow %s\n", old_remote_any > 0 ? "all" : "@LOCAL");
	}
      }`

@zdohnal for the above part what should be the comment that I need to write in conf/cupsd.conf.in ?

Reply:
For example:

```
# Access to the server root (/):
# - default action defined by 'Order', see 'man cupsd.conf'
# - use 'Allow'/'Deny' for configuring access
# - allowing access is required for printer sharing or remote administration
```